File: tests/dynamic_fixtures/go/cmdi_positive.go
// Command injection — positive fixture.
// Vulnerable: passes user input to /bin/sh -c.
// Entry: RunPing(host string)  Cap: CODE_EXEC
// Expected verdict: Confirmed ("; echo NYX_PWN_CMDI" echoes the marker)

package entry

import (
	"fmt"
	"os/exec"
)

func RunPing(host string) {
	fmt.Print("__NYX_SINK_HIT__\n")
	cmd := exec.Command("/bin/sh", "-c", "/bin/echo hello "+host)
	out, _ := cmd.CombinedOutput()
	fmt.Print(string(out))
}

File: tests/dynamic_fixtures/java/cmdi_positive.java
// Command injection — positive fixture.
// Vulnerable: passes user input to /bin/sh -c via Runtime.exec.
// Entry: Entry.runPing(String)  Cap: CODE_EXEC
// Expected verdict: Confirmed ("; echo NYX_PWN_CMDI" echoes the marker)

import java.io.*;

public class Entry {
    public static void runPing(String host) throws Exception {
        System.out.print("__NYX_SINK_HIT__\n");
        String[] cmd = {"/bin/sh", "-c", "echo hello " + host};
        Process p = Runtime.getRuntime().exec(cmd);
        BufferedReader reader = new BufferedReader(new InputStreamReader(p.getInputStream()));
        String line;
        while ((line = reader.readLine()) != null) {
            System.out.println(line);
        }
        p.waitFor();
    }
}
